app/src/main/java/rick_and_morty/ui/episodes/EpisodesViewModel.kt

```diff
@@ -28,21 +35,49 @@ class EpisodesViewModel @Inject constructor(
         _episodes.update { it.copy(isLoading = true) }
         viewModelScope.launch {
             try {
-                val getEpisodes = episodesRepository.getEpisodes(page)
+                var dbEpisodes = getEpisodesFromDatabase()
+                if (dbEpisodes.isEmpty()) {
+                    val apiEpisodes = episodesRepository.getEpisodes(page)
+                    saveEpisodesToDatabase(apiEpisodes)
+                    dbEpisodes = getEpisodesFromDatabase()
+                    page += 1
+                } else if (dbEpisodes.size < page * 20) {
+                    val apiEpisodes = episodesRepository.getEpisodes(page)
+                    saveEpisodesToDatabase(apiEpisodes)
+                    dbEpisodes = getEpisodesFromDatabase()
+                    page += 1
+                }
```

Comment:
The 2 branches of the `if-clause` are exactly the same. Why do we need it?

Reply:
I simplified it